src/main/scala/mesosphere/util/monitor/HeartbeatMonitor.scala

```diff
@@ -0,0 +1,81 @@
+package mesosphere.util.monitor
+
+import javax.inject.Inject
+
+import akka.actor._
+import akka.event.LoggingReceive
+import scala.concurrent.duration._
+
+// HeartbeatActor monitors the heartbeat of some process and executes handlers for various conditions.
+// If an expected heartbeat is missed then execute an `onSkipped` handler.
+// If X number of subsequent heartbeats are missed then execute an `onFailure` handler.
+// Upon creation the actor is in an inactive state and must be sent an EventActivate message to activate.
+// Once activated the actor will monitor for EventPulse messages (these are the heartbeats).
+// The actor may be deactivated by sending it an EventDeactivate message.
+class HeartbeatActor @Inject() (
```

Comment:
pretty sure that the first commit's integration test failed before I added this `Inject`. the logs should still be on TC and I haven't yet squashed the commits, so you should be able to get to it pretty easily.